liberty/LibertyReader.cc

```diff
@@ -3970,8 +3970,15 @@ LibertyReader::seqPortNames(LibertyGroup *group,
     else if (i == 1)
       out_inv_name = value->stringValue();
     else if (i == 2) {
-      size = static_cast<int>(value->floatValue());
-      has_size = true;
+      if (value->isFloat()) {
```

Comment:
I think this would be cleaner if it was rewritten to use switch on the size of the params instead of a for loop. There are really only a few valid cases.

Reply:
Agreed. Seems there are three valid cases:
- 2 values: treat as strings for `out_name` and `out_inv_name`
- 3 values:
  - if last value is int, treat first two strings as `out_name` and `out_inv_name`, and last string as size (for `ff_bank` and `latch_bank`)
  - else, ignore first string, treat last two strings as `out_name` and `out_inv_name`

Reply:
yes